server/sockets.js

```diff
@@ -96,6 +97,28 @@ function socketConnection(socket) {
 			return;
 		}
 
+		var boardData;
+		if (message.data.type === "doc") {
+			boardData = await getBoard(boardName);
+
+			if (boardData.existingDocuments >= config.MAX_DOCUMENT_COUNT) {
+				console.warn("Received too many documents");
+				return;
+			}
+
+			if (message.data.data.length > config.MAX_DOCUMENT_SIZE) {
+				console.warn("Received too large file");
+				return;
+			}
+
+			boardData.existingDocuments += 1;
+		} else if (message.data.type === "delete") {
+			boardData = await getBoard(boardName);
+
+			if (boardData.board[message.data.id].type === "doc") {
+				boardData.existingDocuments -= 1;
+			}
+		}
```

Comment:
This should probably be handled in the BoardData class

Reply:
Also, we should also verify the provided URL, to check that it's a data URL.

Reply:
we would still broadcast the large files to all users in that case

Reply:
Yes, you are right, I hadn't thought about that. I still think the board-specific code should happen in BoardData, but we should probably set the [`httpBufferSize`](https://socket.io/docs/server-api/#new-Server-port-options) option of socket.io in addition, to avoid having to handle and dispatch large messages.